tests/syntax/test_parser.py

```diff
@@ -177,6 +177,222 @@ class C:
+                [Pass()],
+            ):
+                assert True
+            case _:
+                assert False
+
+    def test_parameter(self):
+        mod = parse_string_helper(
+            """
+            behavior test(x, y = "hello"):
+                pass
+        """
+        )
+        stmt = mod.body[0]
+        match stmt:
+            case BehaviorDef(
+                "test",
+                arguments(args=[arg("x"), arg("y")], defaults=[Constant("hello")]),
+                None,
+                [],
+                [Pass()],
+            ):
+                assert True
+            case _:
+                assert False
+
+    def test_precondition(self):
+        mod = parse_string_helper(
+            """
+            behavior test:
+                precondition: True
+                pass
+        """
+        )
+        stmt = mod.body[0]
+        match stmt:
+            case BehaviorDef(
+                "test",
+                arguments(),
+                None,
+                [Precondition(Constant(True))],
+                [Pass()],
+            ):
+                assert True
+            case _:
+                assert False
+
+    def test_invariant(self):
+        mod = parse_string_helper(
+            """
+            behavior test:
+                invariant: True
+                pass
+        """
+        )
+        stmt = mod.body[0]
+        match stmt:
+            case BehaviorDef(
+                "test",
+                arguments(),
+                None,
+                [Invariant(Constant(True))],
+                [Pass()],
+            ):
+                assert True
+            case _:
+                assert False
+
+    def test_precondition_and_invariant(self):
+        mod = parse_string_helper(
+            """
+            behavior test:
+                precondition: True
+                invariant: True
+                precondition: True
+                pass
+        """
+        )
+        stmt = mod.body[0]
+        match stmt:
+            case BehaviorDef(
+                "test",
+                arguments(),
+                None,
+                [
+                    Precondition(Constant(True)),
+                    Invariant(Constant(True)),
+                    Precondition(Constant(True)),
+                ],
+                [Pass()],
+            ):
+                assert True
+            case _:
+                assert False
+
+    def test_invalid_precondition(self):
+        with pytest.raises(SyntaxError) as e:
+            parse_string_helper(
+                """
+                behavior test:
+                    hello()
+                    precondition: True
+                """
+            )
+
+    def test_invalid_invariant(self):
+        with pytest.raises(SyntaxError) as e:
+            parse_string_helper(
+                """
+                behavior test:
+                    hello()
+                    precondition: True
```

Comment:
Should be `invariant`

Reply:
Fixed in 3c42270a0210e559af93795194268623f77c8b48